lib/src/main/java/io/ably/lib/transport/Hosts.java

```diff
@@ -60,9 +60,19 @@ public Hosts(String primaryHost, String defaultHost, ClientOptions options) thro
             setPrimaryHost(defaultHost);
         }
         fallbackHostsUseDefault = options.fallbackHostsUseDefault;
-        if (options.fallbackHosts == null) {
-            fallbackHosts = Arrays.copyOf(Defaults.HOST_FALLBACKS, Defaults.HOST_FALLBACKS.length);
-            fallbackHostsIsDefault = true;
+        if (options.tlsPort != 0 || options.port != 0) {
```

Comment:
Is default value for tlsPort & port 0 🤔 ? Ideally, it should be either null or standard values.

Reply:
Generally, port 0 is reserved, so it shouldn't cause any problem...maybe we can add a minor check for port values to be positive and between 1 to 65535 ( as a part of client-side error handling )

Reply:
I agree that it would be better to have null as the default value, but this is how it was defined [in the code](https://github.com/ably/ably-java/blob/b69cfe70d403b022546cea6123528552319e24b5/lib/src/main/java/io/ably/lib/types/ClientOptions.java#L78). If I change it now then it can break code in clients' libraries.